src/app/page/Recipient/Dashboard.tsx

```diff
@@ -28,31 +28,41 @@ const useStyles = makeStyles((theme) => ({
   },
 }));
 
-const stepMap = (path) =>
+const stepMap = (path: any) =>
   ({
     [routes.recipient.dashboard.submitted]: 0,
     [routes.recipient.dashboard.completed]: 1,
   }[path] || 0);
 
+const sortByDate = (a: MissionInterface, b: MissionInterface) =>
+  new Date(b.createdDate).getTime() - new Date(a.createdDate).getTime();
+
 const completedStatus = [MissionStatus.delivered, MissionStatus.succeeded];
 
 export default function () {
   const classes = useStyles();
   const location = useLocation();
+  // @ts-ignore no types for state
   const auth = useSelector((state) => state.firebase.auth);
-  const [missions, setMissions] = useState({ submitted: [], completed: [] });
+  const [missions, setMissions] = useState<{
+    submitted: MissionInterface[];
+    completed: MissionInterface[];
+  }>({ submitted: [], completed: [] });
 
   useEffect(() => {
     User.getAllRequestedMissions(auth.uid).then((missions) => {
-      let submitted = [];
-      let completed = [];
+      let submitted: MissionInterface[] = [];
+      let completed: MissionInterface[] = [];
 
       missions.forEach((mission) => {
         completedStatus.includes(mission.status)
           ? completed.push(mission)
           : submitted.push(mission);
       });
 
+      submitted.sort(sortByDate);
```

Comment:
you could sort the missions first, then both submitted and completed would be sort by default when they are pushed in